src/rules/no-explicit-type-exports.ts

```diff
@@ -92,18 +132,19 @@ export = {
         node.specifiers.forEach(
           (specifier: TSESTree.ExportSpecifier | TSESTree.ImportClause) => {
             const { name } = specifier.local;
-            let exportedName = name;
+            let aliasName = name;
             if (isExport(specifier)) {
-              exportedName = specifier.exported.name;
-            }
-            if (AllTypedImports.includes(name) || AllTypedImports.includes(`${name} as ${exportedName}`)) {
-              if (name === exportedName) {
-                typedExports.push(name);
-              } else {
-                typedExports.push(`${name} as ${exportedName}`);
+              if (specifier.exported.name !== name) {
+                aliasName = `${name} as ${specifier.exported.name}`;
               }
+            }
+            const isTyped =
+              findIn(name, AllTypedImports) || typeDeclarations.has(name);
+            const isVariable = variableDeclarations.has(name);
+            if (isTyped && !isVariable) {
+              typedExports.push(aliasName);
```

Comment:
Explicitly check that it's not a variable to make sure the in-text lookup doesn't flag a false positive. I'm sure there's a less hacky way to do this but there's only so much time in the day. 😅